src/Services/ComposerConflictsRenovator.php

```diff
@@ -0,0 +1,258 @@
+                            static::COMPOSER_JSON_PATH,
+                            static::REPO_DEFAULT_BRANCH_NAME
+                        ),
+                        $branch_name
+                    );
+                    $this->github_api_controller->createPullRequest(
+                        static::REPO_DEFAULT_BRANCH_NAME,
+                        $branch_name,
+                        $commit_message,
+                        sprintf(
+                            "According to [Wordfence](https://www.wordfence.com/threat-intel/vulnerabilities/), %1\$s %2\$s has a %3\$s CVSS security vulnerability\n\nI'm bumping versions to %4\$s\n\nReferences: %5\$s",
+                            $software_type,
+                            $software_name,
+                            $cvss,
+                            $updated_result->getConflictVersionsString() ?? '',
+                            implode(' , ', $software[0]['references'] ?? [])
+                        )
+                    );
+                } catch (Exception $e) {
+                    continue;
+                }
+            }
+
+            sleep($pause);
+        }
+    }
+
+    /**
+     * Updates composer.json conflict section for specified vulnerability.
+     *
+     * @param array $entry
+     * @param array $composer_json_content
+     *
+     * @return ConflictSectionUpdateResult|null
+     */
+    protected function updateConflictsForVulnerability(
+        array $entry,
+        array $composer_json_content
+    ): ?ConflictSectionUpdateResult {
+        $updated  = false;
+        $software = $entry['software'];
+        if (empty($software)) {
+            return null;
+        }
+
+        foreach ($software as $software_entry) {
+            $affectedVersions = $software_entry['affected_versions'] ?? [];
+            if (!is_array($affectedVersions) || empty($affectedVersions)) {
+                continue;
+            }
+
+            $entry_type = $software_entry['type'];
+            switch ($entry_type) {
+                case 'plugin':
+                case 'theme':
+                    $vulnerability_key =
+                        strtolower(sprintf('wpackagist-%1$s/%2$s', $entry_type, $software_entry['slug']));
+
+                    break;
+
+                case 'core':
+                    if ($software_entry['slug'] !== 'wordpress') {
+                        break;
+                    }
+
+                    $vulnerability_key = 'roots/wordpress';
+
+                    break;
+
+                default:
+                    break;
+            }
+
+            if (!isset($vulnerability_key)) {
+                continue;
+            }
+
+            foreach ($affectedVersions as $affected_version) {
+                $conflict_versions_string = $this->getConflictStringForAffectedVersion($affected_version);
+                if (!isset($conflict_versions_string)) {
+                    continue;
```

Comment:
consider throwing or logging such cases as this should not happen.

Reply:
Yeah, probably I need some help here as I don't know where to log it (while I'm agree with you) as this script is presumably runs on github CI

Reply:
Log it to the stdout/stderr, you will see it in the github action log.

Reply:
Here's an example with the logger as well as nicer exceptions

```php
<?php

declare(strict_types=1);

use Apix\Log\Format\ConsoleColors;
use Apix\Log\Logger;
use Apix\Log\Logger\File;
use Apix\Log\Logger\Stream;
use NunoMaduro\Collision\Provider as ExceptionHandler;

require_once __DIR__ . '/../../vendor/autoload.php';

(new ExceptionHandler())
    ->register();

$stdout = (new Stream('php://stdout'))
    // intercept logs that are >=
    ->setMinLevel('debug')
    // propagate to other loggers
    ->setCascading(true)
    ->setFormat(new ConsoleColors());

$file = (new File(Helper::storageDir() . '/jobs/logs/' . date('Y-m-d') . "-{$name}.log"))
        ->setMinLevel('info')
        // propagate to other loggers
        ->setCascading(true)
        // postpone/accumulate logs processing
        ->setDeferred(false);

$logger = new Logger([$stdout, $file]);

if (\PHP_SAPI !== 'cli') {
    throw new Exception('script must be run from cli');
}
```

```json
    "require": {
        "8ctopus/apix-log": "^3.0",
        "nunomaduro/collision": "^8.0"
    },
```